enhancements/network/communication-flows-matrix-ingress.md

```diff
@@ -0,0 +1,282 @@
+    "nodeRole": "master",
+    "optional": true
+  },
+  {
+    "direction": "ingress",
+    "protocol": "TCP",
+    "port": 51035,
+    "service": "rpc.statd",
+    "nodeRole": "master",
+    "optional": true
+  }
+]
+```
+
+The source of truth for cluster traffic used in the new module is the `EndpointSlices` 
+resource. `EndpointSlices` group network endpoints together by unique combinations of 
+protocol, port number, and `Service` name. For that to work, every application that 
+attracts ingress traffic running in the cluster should have a `Service` object,
+the control plane automatically creates `EndpointSlices` for any `Service` that 
+has a selector specified. `EndpointSlices` can also be directly created when needed
+or managed by other entities or controllers.
+
+To generate the communication matrix the `EndpointSlices` in the cluster should 
+be filtered by `Service` type `NodePort` and `LoadBalancer` as there are also 
+`EndpointSlices` in the cluster for internal `Services`. The node hosted services 
+such as sshd, rpc, etc. are currently missing `EndpointSlices` (see [Open Questions](#open-questions)).
+
+More info about the Kubernetes `EndpointSlices` resource and management can be found
+in https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/#management
+
+Additional information can be added to the communication matrix to describe the use 
+of the ports, for example traffic flows from master-to-master or from node-to-master
+(an API change is required, could be indicated with annotations on the `Service`).
+
+Periodic tests will be added to `openshift-tests` to validate an up-to-date 
+generated communication matrix matches the communication matrix documented
+in OpenShift release documents.
+
+Another test will be added to validate the ports in a generated communication 
+matrix match a snapshot of the node's listening ports (created with the Linux
+`ss` utility).
+
+The communication matrix module will be used to create and apply firewall rules, 
+run E2E `openshift-tests` and upgrade the OpenShift version to validate the rules 
+are acceptable by the current and next OCP release.
+
+A user will be able to run `openshift-tests` or a new `oc` command, 
+`oc adm communication-matrix validate`, to validate the `EndpointSlices` 
+in the cluster match a current snapshot of the node's listening ports.
+
+### Risks and Mitigations
+
+1. Supporting egress traffic flows (out of scope) is more challenging than ingress as 
+   there is no source of truth object such as the `EndpointSlices` for ingress traffic.
+
+2. For egress flows support an API change might be needed, a change that can affect the 
+   work done for ingress flows and should be reviewed and agreed with the upstream Kubernetes
+   community.
+
+3. Generating the communication matrix on MicroShift and/or in AdminNetworkPolicy format is 
+   not supported yet in the current commatrix module.
+
+### Drawbacks
+N/A
+
+## Open Questions
```

Comment:
Add: "Right now the installer has its own hard-coded list of ports to open on different clouds. Can we eventually auto-generate this as well?"